kafka-plugins-0.10/src/main/java/io/cdap/plugin/source/KafkaConfig.java

```diff
@@ -187,6 +195,21 @@ public Integer getMaxRatePerPartition() {
     return maxRatePerPartition;
   }
 
+  @Nullable
+  public Long getDefaultInitialOffset() {
+    if (!containsMacro(initialOffset) && !Strings.isNullOrEmpty(initialOffset)) {
+      if (!initialOffset.equals(OFFSET_START_FROM_SPECIFIC_OFFSET)) {
+        if (initialOffset.equals(OFFSET_START_FROM_BEGINNING)) {
+          return -1L;
+        }
+        if (initialOffset.equals(OFFSET_START_FROM_LAST_OFFSET)) {
+          return -2L;
+        }
+      }
+    }
+    return defaultInitialOffset;
```

Comment:
If defaultInitialOffset is a macro, we'll return null here. Not sure if that will cause issues.
Can you please make you test with a pipeline with macros? Thanks!

Reply:
> If defaultInitialOffset is a macro, we'll return null here. Not sure if that will cause issues.
> Can you please make you test with a pipeline with macros? Thanks!

We did not touch ```defaultInitialOffset``` in order not break backwards compatibility - the following error is thrown when it's set as macro and left as null:
![image](https://user-images.githubusercontent.com/7079754/112294816-838bf200-8c93-11eb-8a08-6e351afb99e3.png)

Reply:
Before your changes, the plugin did support setting topic and offsets as macro params. We need to ensure that this continues to work.
![Screen Shot 2021-03-30 at 10 44 36 AM](https://user-images.githubusercontent.com/2873805/113033001-5df46200-9145-11eb-914b-915a1ebd1f5f.png)

Reply:
We tested with macros and plugin works - here are screenshots of successful pipelines:
![image](https://user-images.githubusercontent.com/7079754/113561504-fdfa3180-9604-11eb-8007-8d0e39b52e7a.png)
![image](https://user-images.githubusercontent.com/7079754/113561536-08b4c680-9605-11eb-9e5e-10c987e1a873.png)

Reply:
Thanks for testing this.